pkgs/by-name/pk/pkl/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  gradle,
+  temurin-bin-21,
```

Comment:
I think it would be preferred to patch out the overly specific JDK requirement. Unless there is a compelling reason not to.

Reply:
https://github.com/NixOS/nixpkgs/pull/402086#issuecomment-2833470424

Reply:
Yes, I saw that - usually upstream specifies whatever JDK works, but doesn't really care. Though in this case it looks like a real mess to patch out so I suppose it's okay.

Reply:
I agree with you but in that case it would be annoying to fix

Reply:
> Though in this case it looks like a real mess to patch out so I suppose it's okay.

FWIW the temurin bins stand on the top of the JDK food chain for linux in nixpkgs as they are used to bootstrap source builds...

But then also I cannot even run `nix-shell -p temurin-bin-21 --run "./gradlew tasks"` in their tree because then it wants a JDK 17 as well... I don't understand why but they are very picky about their JDKs... And even outside of a nix build environment the auto-download doesn't seem to work correctly...